packages/report-server/src/reportBuilder/context/buildContext.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { MicroServiceRequestContext } from '@tupaia/server-boilerplate';
+import { getUniqueEntries } from '@tupaia/utils';
+
+import { FetchResponse } from '../fetch';
+import { extractContextProps } from './extractContextProps';
+import { Context, ContextProp } from './types';
+
+export type ReqContext = {
+  hierarchy: string;
+  services: MicroServiceRequestContext['services'];
+};
+
+type ContextBuilder<K extends keyof Context> = (
+  reqContext: ReqContext,
+  data: FetchResponse,
+) => Promise<Context[K]>;
+
+const isEventResponse = (data: FetchResponse) => data.results.some(result => 'event' in result);
+
+const buildOrgUnits = async (reqContext: ReqContext, data: FetchResponse) => {
+  const orgUnitCodes = isEventResponse(data)
+    ? data.results.map(d => d.orgUnit)
+    : data.results.map(d => d.organisationUnit);
```

Comment:
It really bugs me that events use `orgUnit` and analytics use `organisationUnit`! Not saying we do this now, but I wonder how hard it'd be to migrate events to be consistent...

Reply:
I agree with you we should definitely do this at some point!